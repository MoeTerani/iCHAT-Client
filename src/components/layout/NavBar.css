/* Navbar */
.navbar {
  display: flex;
  justify-content: space-around;
  align-items: center;
  padding: 0.7rem 2rem;
  position: fixed;
  z-index: 1;
  width: 100%;
  height: 60px;
  top: 0;
  border-bottom: solid 1px var(--primary-color);
  opacity: 0.8;
}

.navbar ul {
  display: flex;
}

.navbar a {
  color: #fff;
  padding: 0.45rem;
  margin: 0 0.25rem;
  text-decoration: none;
}
.navbar span {
  color: #fff;
}
.navbar i {
  color: #fff;
}

.navbar a:hover {
  color: var(--primary-color);
}

.navbar .welcome span {
  margin-right: 0.6rem;
}

.btn-dark,
.bg-dark,
.badge-dark,
.alert-dark {
  background: var(--navBg);
  color: #fff;
}

.slider {
  background-color: #999;
  transition: 0.4s;
  border-radius: 34px;
  height: 12px;
  width: 12px;
  display: inline-block;
  position: relative;
}

input:checked + .slider {
  transform: translateX(8px);
}
/*  DARK AND LIGHT THEME END*/

.switch {
  width: 55px;
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.switch div {
  position: relative;
  display: block;
  background: #eee;
  width: 30px;
  border-radius: 50px;
  padding: 0 5px;
  box-sizing: border-box;
  cursor: pointer;
}

.fa-adjust {
  transform: rotate(180deg);
}

.switch input {
  display: none;
}
.dot {
  display: none;
}
.icon--color {
  color: #fff !important;
}

@media (max-width: 1440px) {
  .navbar {
    justify-content: space-between;
  }
}
