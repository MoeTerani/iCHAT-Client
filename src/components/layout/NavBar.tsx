import React from 'react';
import { Link } from 'react-router-dom';
import { FaSun } from 'react-icons/fa';
import { BsFillChatDotsFill } from 'react-icons/bs';
import './NavBar.css';

interface Props {}

const NavBar = (props: Props) => {
  return (
    <nav className='navbar bg-dark'>
      <h1>
        <Link to='/'>
          iCHAT <BsFillChatDotsFill className='icon--color' />
        </Link>
      </h1>
      <div>
        <label className='switch'>
          <FaSun className='icon--color' />
          <div>
            <input
              onChange={() => {
                document.body.classList.toggle('light-theme');
              }}
              type='checkbox'
              className='dot'
            />
            <span className='slider round'></span>
          </div>
        </label>
      </div>
    </nav>
  );
};

export default NavBar;
