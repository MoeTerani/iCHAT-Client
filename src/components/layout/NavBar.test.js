import { shallow } from 'enzyme';
import React from 'react';
import NavBar from './NavBar';

describe('NavBar components', () => {
  test('it renders without errors', () => {
    expect(shallow(<NavBar />)).toMatchSnapshot();
  });
});
