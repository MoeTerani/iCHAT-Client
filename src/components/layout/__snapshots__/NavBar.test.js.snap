// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar components it renders without errors 1`] = `ShallowWrapper {}`;
